src/providers/wms/qgswmsdataitems.cpp

```diff
@@ -98,7 +98,7 @@ QVector<QgsDataItem *> QgsWMSConnectionItem::createChildren()
       QString pathName = layerProperty.name.isEmpty() ? QString::number( layerProperty.orderId ) : layerProperty.name;
       QgsDataItem *layer = nullptr;
 
-      if ( layerProperty.name.isEmpty() )
+      if ( layerProperty.name.isEmpty() || !layerProperty.layer.isEmpty() )
```

Comment:
According to the spec (http://portal.opengeospatial.org/files/?artifact_id=14416, 7.2.4.6.2) 

> A <Title> is mandatory for all layers;

so `!layerProperty.layer.isEmpty()` should always be true, right?

Reply:
> so !layerProperty.layer.isEmpty() should always be true, right?
How will it always be true?

Reply:
@pathmapper if I have correctly understand you, `!layerProperty.layer.isEmpty()` checks if there any nested layers and not titles.

Reply:
@Samweli ahh right, sorry, I was focused on names and titles...

Reply:
ok @pathmapper